cmd/collector/app/server/http_test.go

```diff
@@ -74,7 +74,9 @@ func TestSpanCollectorHTTP(t *testing.T) {
 		Logger:           logger,
 	}
 
-	server, _ := params.ToServer(context.Background(), nil, component.TelemetrySettings{}, nil)
+	server, _ := params.ToServer(context.Background(), nil, component.TelemetrySettings{
+		Logger: logger,
```

Comment:
@yurishkuro As a result of https://github.com/open-telemetry/opentelemetry-collector/issues/11820, a nil pointer reference happens everytime logger is not specified now due to this line. Does that make sense? I would've expected the code to still function if a logger is not supplied by defaulting to a logger.

```go
	errorLog, err := zap.NewStdLogAt(settings.Logger, zapcore.ErrorLevel)
	if err != nil {
		return nil, err // If an error occurs while creating the logger, return nil and the error
	}
```

Reply:
I think it's fine, in OTEL itself logger is never nil and it's a reasonable expectation that all telemetry settings are not nil. We just didn't set them all in the past, but we did have to occasionally add new ones.